exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/project/ProjectRecordBatch.java

```diff
@@ -146,6 +159,27 @@ protected IterOutcome doWork() {
           if (next == IterOutcome.OUT_OF_MEMORY) {
             outOfMemory = true;
             return next;
+          } else if (next == IterOutcome.NONE) {
+            // since this is first batch and we already got a NONE, need to set up the schema
+
+            //allocate vv in the allocationVectors.
+            for (final ValueVector v : this.allocationVectors) {
```

Comment:
The doAlloc() was calling incoming.getRecordCount() which would fail for empty batches, so I did not use it but I can modify doAlloc to take the count parameter and have everyone call the modified version.